### PR TITLE
Extension to support patching in 19.3 docker images

### PR DESCRIPTION
**Pull request by rishabh20oracle**
1. Added a folder named extensions which could be used to provide additional features in docker images and the user can extend the base image to include these features.
2. Added script buildExtensions.sh which provides the user to extend the image and choose which features needs to be included in the extended image.
3. Added a folder called patching under extensions folder which provided patching support.

### CODE REVIEW

**Comment by rishabh20oracle** [comment]
Opened issue #1696

File: OracleDatabase/SingleInstance/extensions/patching/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# PATCH APPLICATION
+
+Applies patches on oracle home. Multiple one-offs can be applied in a single build but only 1 RU can be applied.
+
+Download the release update and one-offs and place them under extensions/patching/patches directory inside subfolders release_update and one_offs respectively.
+
+Once the patches have been placed in the correct directories, use the buildExtensions.sh script to build the extended image with patch support.
```
Comment (yunus-qureshi) [inline OracleDatabase/SingleInstance/extensions/patching/README.md:7]: Can we explain how to use this patched image? and how will datafiles get patched?

Comment (Djelibeybi) [reply OracleDatabase/SingleInstance/extensions/patching/README.md:7]: This pull request has not been merged yet. Once it has, it includes documentation on how the image gets patched.

Comment (yunus-qureshi) [reply OracleDatabase/SingleInstance/extensions/patching/README.md:7]: I would expect that documentation in README.md as part of this PR itself.

Comment (Djelibeybi) [reply OracleDatabase/SingleInstance/extensions/patching/README.md:7]: It is. There is a `README.md` included in this PR.

**Review by latakannan — COMMENTED** [review]
Changes look good for support of RU patching and one off patching

**Review by gvenzl — APPROVED** [review]

**Pull request merged**